omegaconf/basecontainer.py

```diff
@@ -88,7 +88,19 @@ def __repr__(self) -> str:
 
     # Support pickle
     def __getstate__(self) -> Dict[str, Any]:
-        return self.__dict__
+        from omegaconf import DictConfig
+
+        dict_ = copy.copy(self.__dict__)
+        if sys.version_info < (3, 7, 0):  # pragma: no cover
+            ref_type = dict_["_metadata"].ref_type
+            if isinstance(self, DictConfig):
+                if ref_type is not Any:
+                    dict_["_metadata"].ref_type = Dict
+            else:
+                if ref_type is not Any:
+                    dict_["_metadata"].ref_type = List
+
```

Comment:
Please explain.
looks like you are changing the ref_type on serialization. I don't see why this makes sense.